doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -425,16 +425,93 @@ Displaying information about notations
+     punctuation characters.
+   - `IDENT "…"` - a literal string that has the form of an :token:`ident`
+   - `OPT element` - optionally include `element` (e.g. a nonterminal, IDENT "…" or "…")
+   - `LIST1 element` - a list of one or more `element`\s
+   - `LIST0 element` - an optional list of `element`\s
+   - `LIST1 element SEP sep` - a list of `element`\s separated by `sep`
+   - `LIST0 element SEP sep` - an optional list of `element`\s separated by `sep`
+   - `[ elements1 | elements2 | … ]` - alternatives (either `elements1` or `elements2` or …)
+
+   Nonterminals can have multiple **levels** to specify precedence and associativity
+   of its productions.  This feature of grammars makes it simple to parse input
+   such as `1+2*3` in the usual way as `1+(2*3)`.  However, most nonterminals have a single level.
+
+   For example, this output from `Print Grammar tactic` shows the first 3 levels for
+   `tactic_expr`, designated as "5", "4" and "3".  Level 3 is right-associative,
+   which applies to the productions within it, such as the `try` construct::
+
+     Entry tactic:tactic_expr is
+     [ "5" RIGHTA
+       [ tactic:binder_tactic ]
+     | "4" LEFTA
+       [ SELF; ";"; tactic:binder_tactic
+       | SELF; ";"; SELF
+       | SELF; ";"; tactic_then_locality; tactic_then_gen; "]" ]
      | "3" RIGHTA
        [ IDENT "try"; SELF
+       :
+
+   The interpretation of `SELF` depends on its position in the production and the
+   associativity of the level:
+
+   - At the beginning of a production, `SELF` means the next level.  In the
+     fragment shown above, the next level for `try` is "2".  (This is defined by the order
+     of appearance in the grammar or output; the levels could just as well be
+     named "foo" and "bar".)
+   - In the middle of a production, `SELF` means the top level ("5" in the fragment)
+   - At the end of a production, `SELF` means the next level within
+     `LEFTA` levels and the current level within `RIGHTA` levels.
+
+   `NEXT` always means the next level. `nonterminal LEVEL "…"` is a reference to the specified level
+   for `nonterminal`.
+
+   `Associativity <http://camlp5.gforge.inria.fr/doc/htmlc/grammars.html#b:Associativity>`_
+   explains `SELF` and `NEXT` in somewhat more detail.
+
+   The output for `Print Grammar constr` includes :cmd:`Notation` definitions,
+   which are dynamically added to the grammar at run time.
+   For example, in the definition for `operconstr`, the production on the second line shown
+   here is defined by a :cmd:`Reserved Notation` command in `Notations.v`::
+
+     | "50" LEFTA
+       [ SELF; "||"; NEXT
+
+   Similarly, `Print Grammar tactic` includes :cmd:`Tactic Notation`\s, such as :tacn:`dintuition`.
+
+   The file `doc/tools/docgram/fullGrammar` in the source tree extracts the full grammar for
+   |Coq| (not including some optionally-loaded plugins) in a single file with minor changes
```

Comment:
You should probably also specify that it doesn't include notations and tactic notations defined in Coq files since you just mentioned that those would be printed by `Print Grammar`.